File: frontend/components/NetworkSummary.js
import React from 'react';

const NetworkSummary = ({ summary }) => {
  return (
    <div className="network-summary">
      <h2>Network Summary</h2>
      <p><strong>Total Packets:</strong> {summary.total_packets}</p>
      <p><strong>Average Packet Length:</strong> {summary.average_length.toFixed(2)}</p>
      <h3>Top Sources:</h3>
      <ul>
        {summary.top_sources.map((source, index) => (
          <li key={index}>{source[0]} ({source[1]} packets)</li>
        ))}
      </ul>
      <h3>Top Destinations:</h3>
      <ul>
        {summary.top_destinations.map((destination, index) => (
          <li key={index}>{destination[0]} ({destination[1]} packets)</li>
        ))}
      </ul>
    </div>
  );
};

export default NetworkSummary;

File: backend/network_summary.py
from collections import Counter

def generate_network_summary(traffic_data):
    """
    Generate a summary of network traffic data.

    Args:
        traffic_data (list): List of network traffic records.

    Returns:
        dict: A summary of the network traffic, including total packets, average length, 
              top sources, and top destinations.
    """
    total_packets = len(traffic_data)
    if total_packets == 0:
        return {
            "total_packets": 0,
            "average_length": 0,
            "top_sources": [],
            "top_destinations": []
        }

    total_length = sum(traffic['length'] for traffic in traffic_data)
    average_length = total_length / total_packets

    sources = [traffic['source'] for traffic in traffic_data]
    destinations = [traffic['destination'] for traffic in traffic_data]

    top_sources = Counter(sources).most_common(3)
    top_destinations = Counter(destinations).most_common(3)

    return {
        "total_packets": total_packets,
        "average_length": average_length,
        "top_sources": top_sources,
        "top_destinations": top_destinations
    }
